docs/src/tutorials/applications/web_app.jl

```diff
@@ -0,0 +1,165 @@
+# Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors    #src
+# This Source Code Form is subject to the terms of the Mozilla Public License   #src
+# v.2.0. If a copy of the MPL was not distributed with this file, You can       #src
+# obtain one at https://mozilla.org/MPL/2.0/.                                   #src
+
+# # Serving web apps
+
+# This tutorial demonstrates how to setup and serve JuMP models via a REST API.
+
+# In the example app we are building, we solve a trivial mixed-integer program,
+# which is parameterized by the lower bound of a variable. To call the service,
+# users send an HTTP POST request with JSON contents indicating the lower bound.
+# The returned value is the solution of the mixed-integer program as JSON.
+
+# First, we need JuMP and a solver:
+
+using JuMP
+import HiGHS
+
+# We also need [HTTP.jl](https://github.com/JuliaWeb/HTTP.jl) to act as our
+# REST server, and [JSON.jl](https://github.com/JuliaIO/JSON.jl) to marshall
+# data.
+
+import HTTP
+import JSON
+
+# ## The server side
+
+# The core components of our REST server are _endpoints_. These are functions
+# which accept a `Dict{String,Any}` of input parameters, and return a
+# `Dict{String,Any}` as output. The types are `Dict{String,Any}` because we're
+# going to read these to and from JSON.
+
+# Here's a very simple endpoint: it accepts `params` as input, formulates and
+# solves a trivial mixed-integer program, and then returns a dictionary with the
+# result.
+
+function endpoint_solve(params::Dict{String,Any})
+    if !haskey(params, "lower_bound")
+        return Dict{String,Any}(
+            "status" => "failure",
+            "reason" => "missing lower_bound param",
+        )
+    elseif !(params["lower_bound"] isa Real)
+        return Dict{String,Any}(
+            "status" => "failure",
+            "reason" => "lower_bound is not a number",
+        )
+    end
+    model = Model(HiGHS.Optimizer)
+    set_silent(model)
+    @variable(model, x >= params["lower_bound"], Int)
+    optimize!(model)
+    return Dict{String,Any}(
+        "status" => "okay",
+        "terminaton_status" => termination_status(model),
+        "primal_status" => primal_status(model),
+        "x" => has_values(model) ? value(x) : NaN,
```

Comment:
JSON doesn't officially support NaN. This works when reading and writing from Julia, but could be a problem across languages. I think the cleanest workaround is to only set the "x" key in the dictionary if it has a value.

Reply:
I guess it's an okay solution, the user gets a type error if they haven't checked the status before

Reply:
JSON.jl writes NaN out as `"null"`:
```Julia
julia> using JSON

julia> JSON.json(NaN)
"null"
```
but yes, I'll change to only include x if it has a value.